src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs

```diff
@@ -298,5 +306,28 @@ public int LaunchProcessWithDebuggerAttached(TestProcessStartInfo testProcessSta
             this.OnAckMessageRecieved = null;
             return this.dataSerializer.DeserializePayload<int>(ackMessage);
         }
+
+        private ITestCaseEventsHandler GetTestCaseEventsHandler(string runSettings)
+        {
+            ITestCaseEventsHandler testCaseEventsHandler = null;
+
+            if ((XmlRunSettingsUtilities.IsDataCollectionEnabled(runSettings) && DataCollectionTestCaseEventSender.Instance != null) || XmlRunSettingsUtilities.IsInProcDataCollectionEnabled(runSettings))
+            {
+                testCaseEventsHandler = new TestCaseEventsHandler();
+                var testEventsPublisher = testCaseEventsHandler as ITestEventsPublisher;
+
+                if (DataCollectionTestCaseEventSender.Instance != null && XmlRunSettingsUtilities.IsDataCollectionEnabled(runSettings))
+                {
+                    this.outOfProcDataCollectionManager = new ProxyOutOfProcDataCollectionManager(DataCollectionTestCaseEventSender.Instance, testEventsPublisher);
```

Comment:
Not sure if this should be a responsibility of TestRequestHandler to register datacollector managers.